### PR TITLE
UI: Add job version revert buttons

### PR DESCRIPTION
**Pull request by backspace**
This adds a Revert two-step button to the `JobVersions` component for not-current versions, which redirects to the overview on success:

revert-redirect

It checks the job version before and after reversion to mitigate the edge case where reverting to an otherwise-identical version has no effect, as discussed in #10337. Wording tweaks welcome!

no-effect

It uses existing facilities for handling other errors and disabling the button when permissions are lacking.

Implementation notes:

* the pre-and-post reversion job version-checking necessitated adding a watcher for the job record to the versions route to catch version changes originating elsewhere
* to ensure the “No Changes” text and “X Changes” buttons are the same width, I added `.is-fixed-width` to `boxed-section`, would it be better in a new `job-version.scss`? Or elsewhere? 🤔 longing for Tailwind-esque facilities here
* `.has-fading-background` for TSB is very specific to this context, with a white background that fades out to the left 🧐

### CODE REVIEW

**Comment by backspace** [comment]
Re-drafted to add post-reversion redirection to overview, as suggested by @DingoEatingFuzz, which will also likely unintentionally fix the known bug 😆

**Comment by backspace** [comment]
Okay, redirection added, now truly ready for review!

File: ui/app/adapters/job-version.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    const url = addToPath(jobAdapter.urlForFindRecord(jobVersion.get('job.id'), 'job'), '/revert');
+    const [jobName] = JSON.parse(jobVersion.get('job.id'));
+
+    return this.ajax(url, 'POST', {
+      data: {
+        JobID: jobName,
+        JobVersion: jobVersion.number,
+      },
+    });
```
Comment (DingoEatingFuzz) [inline ui/app/adapters/job-version.js:16]: I forget how this URL building stuff works. Is this going to append the namespace as a query param?

**Review by DingoEatingFuzz — CHANGES_REQUESTED** [review]
This looks good! 

The only thing I'd like verification on (and maybe a test) is if the revert request is correct when not in the default namespace.

File: ui/app/adapters/job-version.js (hunk 1 above)
Comment (backspace) [reply ui/app/adapters/job-version.js:16]: ya, I did manually test that but it makes sense to automate, how does c30917f look for that?

Comment (DingoEatingFuzz) [reply ui/app/adapters/job-version.js:16]: Looks good!

**Review by DingoEatingFuzz — APPROVED** [review]

**Pull request merged**